client/components/status/status.js

```diff
@@ -75,6 +75,20 @@ export default class Status extends React.Component {
         const macaddr = searchParams.get(
           captivePortalLoginForm.macaddr_param_name,
         );
+
+        window.addEventListener("message", (event) => {
+          // For security reasons, read https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#security_concern
+          if (event.origin !== new URL(captivePortalLoginForm.action).origin) {
```

Comment:
I would leave the possibility of using also URLs on the same origin as the app itself, not only on the captive portal host.